### PR TITLE
Move args default value to only be run by master

### PR DESCRIPTION
**Pull request by jpa198**
This prevents a crash when running HF

#### Checklist
- Have you updated the README (yes/no)?
- Have you updated the CHANGELOG (yes/no)?

### CODE REVIEW

**Review by jasonmotha — APPROVED** [review]

File: scripts/hf_sim.py [hunk 1]
```diff
@@ -203,8 +203,8 @@ def args_parser(cmd=None):
             # invalid arguments or -h
             comm.Abort()
 
-    if args.sim_bin is None:
-        args.sim_bin = binary_version.get_hf_binmod(args.version)
+        if args.sim_bin is None:
+            args.sim_bin = binary_version.get_hf_binmod(args.version)
```
Comment (jasonmotha) [inline scripts/hf_sim.py:207]: Are you merging with the block above / below too?

File: scripts/hf_sim.py [hunk 2]
```diff
@@ -203,8 +203,8 @@ def args_parser(cmd=None):
...
 
     if is_master:
```
Comment (claudio525) [inline scripts/hf_sim.py:209]: Can make this one `if is_master check`?

File: scripts/hf_sim.py (hunk 1 above)
Comment (jpa198) [reply scripts/hf_sim.py:207]: Done

**Review by jasonmotha — APPROVED** [review]

**Review by shot0829 — APPROVED** [review]

**Review by claudio525 — APPROVED** [review]

**Comment by jasonmotha** [comment]
Just curious. What actual bug does this fix? From what I can see in the code, because args gets broadcast a few lines down all processes should have the same copy of args?

**Comment by jpa198** [comment]
> Just curious. What actual bug does this fix? From what I can see in the code, because args gets broadcast a few lines down all processes should have the same copy of args?

For all processes but rank 0, args = None, therefore args.sim_bin doesn't exist and throws an exception. 
This happens in MPI code, cue Bad Things.

**Pull request merged**